tests/foreman/cli/test_discoveryrule.py

```diff
@@ -0,0 +1,422 @@
+# -*- encoding: utf-8 -*-
+"""Test class for Foreman Discovery Rules
+
+@Requirement: Discoveryrule
+
+@CaseAutomation: Automated
+
+@CaseLevel: Acceptance
+
+@CaseComponent: CLI
+
+@TestType: Functional
+
+@CaseImportance: High
+
+@Upstream: No
+"""
+
+from robottelo.decorators import (
+    run_only_on,
+    tier1,
+    tier2,
+    stubbed,
+)
+from robottelo.test import CLITestCase
+
+
+class DiscoveryRuleTestCase(CLITestCase):
+    """Implements Foreman discovery Rules tests in CLI."""
+
+    @run_only_on('sat')
+    @stubbed
+    @tier1
+    def test_positive_create_with_name(self):
+        """Create Discovery Rule using different names
+
+        @id: 066e66bc-c572-4ae9-b458-90daf83bab54
+
+        @Assert: Rule should be successfully created
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_search(self):
+        """Create Discovery Rule using different search queries
+
+        @id: 2383e898-a968-4183-a270-55e9350e0596
+
+        @Assert: Rule should be successfully created and has expected search
+        value
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_hostname(self):
+        """Create Discovery Rule using valid hostname
+
+        @id: deee22c3-dcfd-4940-b27c-cca137ec9a92
+
+        @Assert: Rule should be successfully created and has expected hostname
+        value
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_by_associating_org_loc_name(self):
+        """Create discovery rule by associating org and location names
+
+        @id: f0d550ae-16d8-48ec-817e-d2e5b7405b46
+
+        @Assert: Rule was created and with given org & location names.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_by_associating_org_loc_id(self):
```

Comment:
btw, as you mentioned `org_loc_id` in name, i assume there's another way, by specifying org and loc names? If so, please also add test stubs for that